flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/metrics/NumberOfFullRestartsGauge.java

```diff
@@ -42,6 +42,6 @@ public NumberOfFullRestartsGauge(ExecutionGraph executionGraph) {
 
 	@Override
 	public Long getValue() {
-		return eg.getNumberOfFullRestarts();
+		return eg.getNumberOfRestarts();
```

Comment:
Can there be problems with respect to memory visibility? Are the reporters running in separate thread pools?

Reply:
It's a good question. The metric reporting happens in a separate thread ("Flink-MetricRegistry"). So dirty read is possible to happen if a metric reports value which is modified in another thread.

But I'm not sure whether we need to guarantee there is no dirty read in metric reporting.
I took a look and find almost all metrics in Flink does not take this into consideration.
And I guess `ExecutionGraph#globalModVersion` was made to be volatile mainly for its version functionality.

WDYT?

Reply:
This isn't a problem.

Reply:
Ok, let's accept that the reported value can be stale.